compiler/src/dotty/tools/dotc/printing/PlainPrinter.scala

```diff
@@ -60,6 +60,8 @@ class PlainPrinter(_ctx: Context) extends Printer {
           homogenize(tp.ref)
         case AppliedType(tycon, args) =>
           tycon.dealias.appliedTo(args)
+        case tp: AppliedTermRef =>
+          homogenize(tp.underlying)
```

Comment:
That seems pretty drastic to me, it means that AppliedTermRef will always be widened before being pretty-printed with -Ytest-pickler, so you may miss differences.